routetable/route_table.go

```diff
@@ -535,16 +770,22 @@ func (r *RouteTable) syncRoutesForLink(ifaceName string) error {
 			continue
 		}
 
-		routeExpected := expectedCIDRs.Contains(dest)
+		expectedTarget, expectedTargetFound := expectedCIDRsToTarget[dest]
+		routeExpected := expectedTargetFound || (r.ipVersion == 6 && dest == ipV6LinkLocalCIDR)
 		var routeProblems []string
 		if !routeExpected {
 			routeProblems = append(routeProblems, "unexpected route")
 		}
 		if !r.deviceRouteSourceAddress.Equal(route.Src) {
 			routeProblems = append(routeProblems, "incorrect source address")
 		}
-		if dest != ipV6LinkLocalCIDR && r.deviceRouteProtocol != route.Protocol {
-			routeProblems = append(routeProblems, "incorrect protocol")
+		if dest != ipV6LinkLocalCIDR {
+			if r.deviceRouteProtocol != route.Protocol {
+				routeProblems = append(routeProblems, "incorrect protocol")
+			}
+			if expectedTargetFound && expectedTarget.RouteType() != route.Type {
+				routeProblems = append(routeProblems, "incorrect type")
+			}
 		}
 		if len(routeProblems) == 0 {
```

Comment:
@robbrockbank It is great to have the check for `RouteType` here as it could useful for egress ip.
Could we also add an extra check for gateway ip? 
```
	if route.Gw.Equal(expectedTarget.GW.AsNetIP()) {
            routeProblems = append(routeProblems, "mismatch gateway")
	}
```